hasher-matcher-actioner/hmalib/common/actioner_models.py

```diff
@@ -158,118 +158,69 @@ def from_match_message_and_label(
+
+    def call(self, _data: str) -> Response:
         return get(self.url)
 
 
 class WebhookPutActionPerformer(WebhookActionPerformer):
-    def call(self, **kwargs) -> Response:
-        data = kwargs.get("data")
+    """Hit an arbitrary endpoint with a PUT"""
+
+    def call(self, data: str) -> Response:
         return put(self.url, data)
 
 
 class WebhookDeleteActionPerformer(WebhookActionPerformer):
-    def call(self, **kwargs) -> Response:
-        return delete(self.url)
-
-
-def get_all_subclasses_rec(recursive: t.Set[t.Type]) -> t.Set[t.Type]:
-    subclasses = {subclass for cls in recursive for subclass in cls.__subclasses__()}
-    union = recursive.union(subclasses)
-    if union != recursive:
-        return get_all_subclasses_rec(union)
-    return recursive
-
-
-# TODO David said he wanted to refactor this
-@dataclass
-class ActionPerformerConfig(config.HMAConfig):
-    """
-    An adapter to use the HMAConfig Class to store differnet types of
-    ActionPerformer configs. Since ActionPerformer configs are all unique by label but may
-    be of different types, we store the name of the concrete type and the
-    attributes of that type in AWS.
+    """Hit an arbitrary endpoint with a DELETE"""
 
-    DO NOT CALL config.HMAConfig FUNCTIONS DIRECTLY
-
-    To load a config from a label use ActionPerformerConfig.get_performer
-
-    To update an action performer ActionPerformerConfig.update_performer
-    """
-
-    concrete_type: str
-    concrete_type_attrs: t.Dict[str, t.Any]
-
-    @classmethod
-    def get_performer(cls, action_label: ActionLabel) -> t.Optional["ActionPerformer"]:
-        """
-        Load from AWS the correct actioner config as the correct concrete class if it exists
-        """
-        name = action_label.value
-        performer_config: t.Optional[ActionPerformerConfig] = cls.get(name)
-        if performer_config:
-            subclasses_classes = {
-                cls.__name__: cls for cls in get_all_subclasses_rec({ActionPerformer})
-            }
-            return subclasses_classes[performer_config.concrete_type](
-                action_label=action_label, **performer_config.concrete_type_attrs
-            )
-        return None
-
-    @classmethod
-    def update_performer(cls, action_performer: ActionPerformer) -> None:
-        """
-        Use this function to update the config in AWS instead of config.update_config directly
-        """
-        attrs = {
-            field.name: getattr(action_performer, field.name)
-            for field in fields(action_performer)
-        }
-        del attrs["action_label"]
-        action_performer_config = ActionPerformerConfig(
-            name=action_performer.action_label.value,
-            concrete_type=action_performer.__class__.__name__,
-            concrete_type_attrs=attrs,
-        )
-        config.update_config(action_performer_config)
+    def call(self, _data: str) -> Response:
+        return delete(self.url)
```

Comment:
For this and GET we will need to find another way of including the match message in the request